src/linux_ip_socket/linux_ip_socket.cpp

```diff
@@ -0,0 +1,336 @@
+            std::cerr << "aborting." << std::endl;
+            abort();
+        }
+
+        for(int descriptor_index = 0; descriptor_index < descriptors_table_count; ++descriptor_index) {
+            if(descriptors_table[descriptor_index].revents & POLLIN) {
+                if(descriptor_index == 0) {
+                    accept_connection(descriptors_table, descriptors_table_count);
+                } else {
+                    read_data_or_disconnect(descriptor_index, descriptors_table, descriptors_table_count);
+                }
+            }
+        }
+    }
+}
+
+void
+linux_ip_socket_private_data::driver_send(const uint8_t* const data, const size_t length)
+{
+    const int sockfd = connect_to_remote_driver();
+    if(sockfd == INVALID_SOCKET_ID) {
+        return;
+    }
+
+    size_t send_buffer_index = 0;
+
+    m_send_buffer[0] = START_BYTE;
+    ++send_buffer_index;
+
+    size_t index = 0;
+    bool escape = false;
+    while(index < length) {
+        if(escape) {
+            m_send_buffer[send_buffer_index] = data[index];
+            ++send_buffer_index;
+            ++index;
+            escape = false;
+        } else if(data[index] == START_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else if(data[index] == ESCAPE_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else if(data[index] == STOP_BYTE) {
+            m_send_buffer[send_buffer_index] = ESCAPE_BYTE;
+            ++send_buffer_index;
+            escape = true;
+        } else {
+            m_send_buffer[send_buffer_index] = data[index];
+            ++send_buffer_index;
+            ++index;
+        }
+
+        if(send_buffer_index >= DRIVER_SEND_BUFFER_SIZE) {
+            send_packet(sockfd, DRIVER_SEND_BUFFER_SIZE);
+            send_buffer_index = 0;
+        }
+    }
+
+    m_send_buffer[send_buffer_index] = STOP_BYTE;
+    ++send_buffer_index;
+    send_packet(sockfd, send_buffer_index);
+    send_buffer_index = 0;
+    close(sockfd);
+}
+
+void
+linux_ip_socket_private_data::find_addresses(addrinfo** target, const char* address, const unsigned int port)
+{
+    addrinfo hints;
+    memset(&hints, 0, sizeof(addrinfo));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+
+    const std::string remote_port = std::to_string(port);
+
+    const int getaddrinfo_result = getaddrinfo(address, remote_port.c_str(), &hints, target);
+
+    if(getaddrinfo_result != 0) {
+        std::cerr << "getaddrinfo returned an error: " << gai_strerror(getaddrinfo_result) << std::endl;
+        std::cerr << "Aborting." << std::endl;
+        abort();
+    }
+}
+
+void
+linux_ip_socket_private_data::parse_recv_buffer(const size_t length)
+{
+    for(size_t i = 0; i < length; ++i) {
```

Comment:
BTW -> if we connect and disconnect for each send.... do we even need escaping? We need to talk.

Reply:
No, it is not needed anymore.

Reply:
Ok, I would like to keep this.